### ISSUE TITLE
SC2096: Detects also non-shebang comments (i.e. `# ...` without `#! ...)

### ISSUE THREAD

**Issue by HenrikBengtsson** [koalaman/shellcheck#2740]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (i.e. https://shellcheck.net/wiki/SC2096)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

With a `test.sh` file as in:

```sh
$ echo "# /path/to/file --arg1 --arg2" > test.sh

$ cat test.sh
# /path/to/file --arg1 --arg2
```

ShellCheck incorrectly thinks this comment is a shebang.

#### Here's what shellcheck currently says:

ShellCheck incorrectly thinks this comment is a shebang;

```sh
$ shellcheck -s bash test.sh
# /path/to/file --arg1 --arg2
^---------------------------^ SC2096 (error): On most OS, shebangs can only specify a single parameter.
^-- SC1113 (error): Use #!, not just #, for the shebang.
```

#### Here's what I wanted or expected to see:

I would not expect SC2096 here since it's not a shebang (as SC1113 suggests).  So I'd only expect:

```sh
$ shellcheck -s bash test.sh
# /path/to/file --arg1 --arg2
^-- SC1113 (error): Use #!, not just #, for the shebang.
```

#### Addition info

FWIW, this happens only if the path is absolute, i.e. it starts with a slash (`/`).  It does not happen with, say,

```sh
# path/to/file --arg1 --arg2
```

Version used:

```sh
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```
